### DISCUSSION TITLE
APK , malware detected

### DISCUSSION THREAD

**Discussion by atmanegara** [apache/cordova#365] [Q&A]
my apk was detected as malware by avast/avg antivirus,
here is my app in playstore
https://play.google.com/store/apps/details?id=bkpsdmd.pemkabhst.brbpresensi
![image](https://user-images.githubusercontent.com/9096711/207885426-0d0cd6a8-c5e3-4fc9-af6a-fbff6d0244a2.png)

information please?
i build with cordova 11,
plugins that I use
<img width="538" alt="image" src="https://user-images.githubusercontent.com/9096711/207883638-8bd2b062-66fd-40a7-b464-876d4e11566c.png">

**Comment by breautek** [comment]
There isn't really enough information here to say the cause. You might have to get into contact with AVG to see if they can provide more details on what or why it considers your app a threat.

It could be one of those plugins, it could be a dependency one of those plugins is importing, it could be a sub-dependency of one of those plugins. The point is, it could be anything. In fact, it could even be a false positive by AVG.

If what it considers a threat is reported as a security vulnerability and has a CVE, `npm audit` may give you a hint to where to look. But without knowing specific details on what AVG is "finding", then you're going on a wild goose chase. NPM Audit is littered with low risk vulnerabilities that is likely doesn't apply to you, or the app, or may only apply to the development machine but not the actual runtime, so there will likely be noise to shift through.

I'd also suggest running `npm upgrade` to update your dependency tree with the latest dependencies that packages uses, before you run `npm audit` which will likely resolve most of any audit issues and reduce that noise.

Since this doesn't describe a bug, I'll be converting this into a Discussion.

**Reply by atmanegara** [reply]
sorry sir, I just tried to rebuild, I tried to scan again with avast and avg, the results are safe and clean, does my apk in playstore allow downloading to be malware?

**Reply by breautek** [reply]
Well, no it's against Google's terms of service to publish malware to the play store, but Avast/AVG is not the authoritative figure on this manner.

In my experience, Google Play will warn you if your application contains known vulnerabilities and may force you to deal with it depending on the severity of the vulnerabilities. You can review their policies [here](https://play.google.com/about/developer-content-policy/), with the [Privacy, Deception, and Device Abuse](https://support.google.com/googleplay/android-developer/topic/9877467) and [Malware](https://support.google.com/googleplay/android-developer/topic/9975838) sections.

So my best advice is to investigate, but don't fret too much if you can't find any reason why AVG is reporting your app. (e.g. there are no open CVEs on the dependencies you have or NPM audit is clean).

**Reply by atmanegara** [reply]
thank you sir, the information helped me a lot
